Modules/_stat.c

```diff
@@ -492,113 +492,140 @@ ST_CTIME\n\
 ");
 
 
+static int
+stat_exec(PyObject *module)
+{
+#define ADD_INT_MACRO(module, macro)                                  \
+    do {                                                              \
+        if (PyModule_AddIntMacro(module, macro) < 0) {                \
```

Comment:
Your PR doesn't work: 

```
./python -c 'import _stat; _stat.S_IFDIR'
AttributeError: module '_stat' has no attribute 'S_IFDIR'
```

You should try:

```suggestion
        if (PyModule_AddIntConstant(module, #macro, macro) < 0) {                \
```